src/monster.cpp

```diff
@@ -1289,6 +1291,12 @@ void monster::die(Creature* nkiller) {
                                   name().c_str() );
         }
     }
+    // We were tied up at the moment of death, add a short rope to inventory
+    if ( has_effect("tied") ) {
```

Comment:
I want to say this should itself have a hallucination check, but realistically you can't have a tied-up hallucination.  So yeah, leave it as is.